py-polars/polars/io/_utils.py

```diff
@@ -145,17 +145,33 @@ def managed_file(file: Any) -> Iterator[Any]:
         if _FSSPEC_AVAILABLE:
             from fsspec.utils import infer_storage_options
 
-            if not has_non_utf8_non_utf8_lossy_encoding:
-                if infer_storage_options(file)["protocol"] == "file":
+            # check if it is a local file
+            if infer_storage_options(file)["protocol"] == "file":
+                # (lossy) utf8
+                if has_utf8_utf8_lossy_encoding:
                     return managed_file(normalise_filepath(file, check_not_dir))
+                # decode first
```

Comment:
Fsspec did not seem to decode, so we use the same logic as when fsspec is not installed.